src/app/qgsmaptooladdpart.cpp

```diff
@@ -194,6 +198,12 @@ void QgsMapToolAddPart::cadCanvasReleaseEvent( QgsMapMouseEvent * e )
       vlayer->endEditCommand();
 
       vlayer->triggerRepaint();
+
+      if (( !isGeometryEmpty ) && QgsWkbTypes::isSingleType( vlayer->wkbType() ) )
```

Comment:
It doesn't work if I do it like that.
This is why I didn't use the dedicated signal from the beginning.
I banged my head around this and the only solution that I see is the current one (I'm a newbie with respect to C++).